src/tuesday/core/plotting/powerspectra.py

```diff
@@ -0,0 +1,389 @@
+    ax.set_xlabel(xlabel, fontsize=fontsize)
+    ax.set_ylabel(ylabel, fontsize=fontsize)
+    if log[0]:
+        ax.set_xscale("log")
+    if log[1]:
+        ax.set_yscale("log")
+    if label is not None:
+        ax.legend(**legend_kwargs)
+    return fig, ax
+
+
+def plot_2d_power_spectrum(  # noqa: C901
+    wavemodes: un.Quantity,
+    power_spectrum: un.Quantity,
+    fig: plt.Figure | None = None,
+    ax: plt.Axes | None = None,
+    title: str | None = None,
+    xlabel: str | None = None,
+    ylabel: str | None = None,
+    clabel: str | None = None,
+    cmap: str | None = "viridis",
+    fontsize: float | None = 16,
+    vmin: float | None = None,
+    vmax: float | None = None,
+    log: list[bool] | None = False,
+    smooth: float | bool = False,
+    cbar: bool | None = True,
+) -> tuple[plt.Figure, plt.Axes]:
+    """
+    Plot a 2D power spectrum.
+
+    Parameters
+    ----------
+    wavemodes : un.Quantity
+        Wavemodes [kperp, kpar].
+        Accepted units are 1/Mpc or h/Mpc.
+    power_spectrum : un.Quantity
+        Power spectrum array of shape [Nsamples, Nkperp, Nkpar]
+        or [Nkperp, Nkpar].
+        There are six accepted units: mK^2 Mpc^3, mK^2 Mpc^3/h^3, mK^2,
+        or dimensionless instead of mK^2.
+    fig : plt.Figure, optional
+        Figure object to plot on. If None, a new figure is created.
+    axs : plt.Axes | list[plt.Axes], optional
+        Axes object(s) to plot on. If None, new axes are created.
+    title : str, optional
+        Title(s) of the plot.
+    xlabel : str, optional
+        Label for the x-axis.
+    ylabel : str, optional
+        Label for the y-axis.
+    clabel : str, optional
+        Label for the colorbar.
+    cmap : str, optional
+        Colormap for the plot.
+    fontsize : float, optional
+        Font size for the plot labels.
+    vmin : float, optional
+        Minimum value for the color scale.
+    vmax : float, optional
+        Maximum value for the color scale.
+    log : list[bool], optional
+        List of booleans to set the kperp, kpar, and PS axes to log scale.
+    smooth : float, optional
+        Standard deviation for Gaussian smoothing.
+        If True, uses a standard deviation of 1.
+    """
+    rcParams.update({"font.size": fontsize})
+    kperp = wavemodes[0]
+    kpar = wavemodes[1]
+    if xlabel is None:
+        if kperp.unit == 1 / un.Mpc:
+            xlabel = r"$k_\perp \, [\rm{Mpc}^{-1}]$"
+        elif kperp.unit == littleh / un.Mpc:
+            xlabel = r"$k_\perp \, [h \, \rm{Mpc}^{-1}]$"
+        else:
+            raise ValueError("kperp must be in units of 1/Mpc or h/Mpc.")
+
+    if ylabel is None:
+        if kpar.unit == 1 / un.Mpc:
+            ylabel = r"$k_\parallel \, [\rm{Mpc}^{-1}]$"
+        elif kpar.unit == littleh / un.Mpc:
+            ylabel = r"$k_\parallel \, [h \, \rm{Mpc}^{-1}]$"
+        else:
+            raise ValueError("kpar must be in units of 1/Mpc or h/Mpc.")
```

Comment:
it might be easier to try using the astropy Latex formatter, and also just to try to coerce the units to the "correct" units, and then raise ValueError otherwise. Maybe people have weird but OK units like `h/km` or something.